model/task_lifecycle.go

```diff
@@ -1913,7 +1914,7 @@ func UpdateBuildAndVersionStatusForTask(ctx context.Context, t *task.Task) error
 			if err != nil {
 				return errors.Wrap(err, "getting context for tracing")
 			}
-			_, span := tracer.Start(traceContext, "version-completion")
+			_, span := tracer.Start(traceContext, "version-completion", trace.WithNewRoot())
```

Comment:
Is it intentional that this doesn't use the context passed back for later operations in the function? If we don't use the context that's returned here, then other operations that pass the context (like `UpdatePatchStatus`) that could be traced with OTel won't be included as part of this span.

Reply:
This returns before calling UpdatePatchStatus. We may want to extend it eventually but it's out of the scope for this ticket.